src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java

```diff
@@ -50,6 +51,63 @@ public void shouldInsertTransaction() {
         assertThat(retrievedTransaction.isLive(), is(true));
     }
 
+    @Test
+    public void shouldInsertTransactionWithSourceCardApi() {
+        TransactionFixture fixture = aTransactionFixture()
+                .withDefaultCardDetails()
+                .withNetAmount(55)
+                .withTotalAmount(55)
+                .withTransactionType("PAYMENT")
+                .withLive(true)
+                .withSource(Source.CARD_API.toString())
+                .withDefaultTransactionDetails();
+        TransactionEntity transactionEntity = fixture.toEntity();
+
+        transactionDao.upsert(transactionEntity);
+
+        TransactionEntity retrievedTransaction = transactionDao.findTransactionByExternalId(transactionEntity.getExternalId()).get();
+
+        assertThat(retrievedTransaction.getSource(), is(Source.CARD_API));
+    }
+
+    @Test
+    public void shouldInsertTransactionWithSourceCardPaymentLink() {
```

Comment:
This feels redundant. Don't think we need to have a test for each Source constant for integration test. Could cover possible combinations in unit tests

Reply:
agree